docs/Course Wiki/ECE Course Offerings/ECE374B.md

```diff
@@ -0,0 +1,62 @@
+- Undecidability
+- NP, NP-Hardness, P vs NP
+
+## Prerequisites
+- [CS 173](../CS%20Course%20Offerings/CS173.md)
+- [CS 225](../CS%20Course%20Offerings/CS225.md)
+
+There are two official prerequisites to ECE 374, CS 173 - Discrete Structures, and CS 225 - Data Structures and Algorithms. 374B expands on the theoretical computer science from CS173 and covers algorithms more in depth than CS225, without any of the programming. For part one of the course, you should remember induction, set notation, and FSMs from CS 173. Induction and recursion are fundamental to the course, so it is important to know them well. In the second portion of the course, there is a significant overlap with the graph algorithms taught in CS225.
+
+## When to Take it
+ECE374 is a required course for all Computer Engineering and Computer Science majors. ECE 374 is not used as a prerequisite for any course in the ECE department, so for students not interested in higher level CS courses, there is little reason to rush it. However, due to the knowledge needed from CS 225 and CS 173, it might help to take it within a few semesters after finishing CS 225. On the other hand, taking this course early would help to prepare for Dynamic Programming interview questions. It is very common to encouter DP questions when applying for software engineering internships, and the HWs in ECE374 help prepare for them well. This course is fairly challenging so take caution if you plan to take this course with too many other technical courses.
+
+## Course Structure
+In terms of content the course is split into three sections, each with their own midterm. The first 5 or so weeks will be spent covering the basics of models of computation. The next 6ish weeks will cover design and analysis of algorithms. And the rest of the course will cover decidability, reductions, and P vs NP.
+
+During the semester the course has 9-10 Homeworks, due every week besides weeks of midterms. You are allowed to work with other students in groups of three. Homeworks include problems that were covered in discussion, so if you go to discussion you know the answer to 1-2 problems on each HW. About 75% of the grade comes from the three midterms and the final, and the rest of the 25% from the HWs. You are allowed to drop the lowest exam, including the final. This means that if you do well on the first three exams, you can completely skip the final. There is also a provided cheat sheet for each exam. This makes memorizing theoretical definitions less important, but it also makes harder questions appear on the exam.
+
+The course has two components, discussions and lecture. Lectures are twice a week and run for 75 minutes. Discussions are also twice a week and 50 minutes.
+
+## Instructors
+In recent years (as of Fall 2023), Proffessor Nickvash Kani has taught the B section of the course. 
+
+## Course Tips
+Go to discussions, make sure to write down the problems so you know them when they come up in the homeworks.
```

Comment:
"Go to discussions and make sure to write down"

Put another line between each sentence so it shows up in the Markdown as different paragraphs.